mcs/class/System.Web/System.Web.Compilation/AppResourcesAssemblyBuilder.cs

```diff
@@ -234,22 +234,11 @@ string SetAlPath (ProcessStartInfo info)
 		{			
 			if (RuntimeHelpers.RunningOnWindows) {
 				string alPath;
-				string monoPath;
-				PropertyInfo gac = typeof (Environment).GetProperty ("GacPath", BindingFlags.Static|BindingFlags.NonPublic);
-                                MethodInfo get_gac = gac.GetGetMethod (true);
-                                string p = Path.GetDirectoryName ((string) get_gac.Invoke (null, null));
-				monoPath = Path.Combine (Path.GetDirectoryName (Path.GetDirectoryName (p)), "bin\\mono.bat");
-                                if (!File.Exists (monoPath)) {
-                                        monoPath = Path.Combine (Path.GetDirectoryName (Path.GetDirectoryName (p)), "bin\\mono.exe");
-					if (!File.Exists (monoPath)) {
-						monoPath = Path.Combine (Path.GetDirectoryName (Path.GetDirectoryName (Path.GetDirectoryName (p))), "mono\\mono\\mini\\mono.exe");
-						if (!File.Exists (monoPath))
-							throw new FileNotFoundException ("Windows mono path not found: " + monoPath);
-					}
-				}
+				string monoPath = MonoExeLocator.MonoPath;
+				string p = MonoExeLocator.GacPath;
 				alPath = Path.Combine (p, framework_version + "\\al.exe");
```

Comment:
I'd prefer to move it so we have all this weird logic in one place, but it's not a big deal.